.github/workflows/benchmark.yaml

```diff
@@ -0,0 +1,160 @@
+name: On Demand Benchmark
+on:
+  issue_comment:
+    types: [created]
+
+jobs:
+  run-benchmark:
+    if: ${{ github.event.issue.pull_request && contains(github.event.comment.body, '/run-benchmark') }}
+    permissions: write-all
+    runs-on: ubuntu-latest
+    steps:
+      # https://github.com/actions/checkout/issues/331#issuecomment-1567674767
+      - id: 'get-branch'
+        run: echo ::set-output name=branch::$(gh pr view $PR_NO --repo $REPO --json headRefName --jq '.headRefName')
+        env:
+          REPO: ${{ github.repository }}
+          PR_NO: ${{ github.event.issue.number }}
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          submodules: recursive
+      - name: Setup Rust
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          override: true
+      - uses: Swatinem/rust-cache@v2
+        with:
+          shared-key: benchmark-tests
+          cache-all-crates: true
+      - name: Run sccache-cache
+        uses: mozilla-actions/sccache-action@v0.0.3
+      - uses: davidB/rust-cargo-make@v1
+      - name: Install Protoc
+        uses: arduino/setup-protoc@v1
+        with:
+          repo-token: ${{ secrets.GITHUB_TOKEN }}
+      - name: Setup Redis
+        uses: shogo82148/actions-setup-redis@v1.29.0
+        with:
+          redis-version: latest
+          auto-start: false
+      - name: Build Benchmark Report Job
+        run: |
+          echo "Building all targets including benchmark-report job"
+          cargo build
+      - name: Run Benchmark
+        run: |
+          echo "Running all benchmark jobs and saving to files"
+      - name: Post results as a comment
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          ISSUE_NUMBER: ${{ github.event.issue.number }}
+          WORKFLOW_ID: ${{ github.run_id }}
+          COMMENT_BODY: ${{ github.event.comment.body }}
+        run: |
+          #! /bin/bash
+          
+          # Create a dummy benchmark result
+          mkdir -p ./benchmark-data-current
+          
+          ./target/debug/benchmark_cold_start_large --quiet --json spawned > benchmark-data-current/benchmark_cold_storage_large.json
```

Comment:
Based on the closed benchmark tickets we should have the following benchmarks:
- Suspend benchmark (https://github.com/golemcloud/golem/issues/384)
- 3 different cold start benchmarks (https://github.com/golemcloud/golem/issues/379)
- 3 different latency benchmarks (https://github.com/golemcloud/golem/issues/382)
- Durability overhead benchmark (https://github.com/golemcloud/golem/issues/403)
- 3 different cold-start benchmarks with compilation cache service enabled (https://github.com/golemcloud/golem/issues/381)
- Throughput benchmark (golem and native) (https://github.com/golemcloud/golem/issues/383)

Some of  these are missing, I hope only from this script ;)

Reply:
I can suspend and throughput. But there will definitely be iteration of 1 or 2 commits after the merge. I tested this only with golem-timeline, lets see how golem-repo is going to be testing my patience :)

Reply:
in relation to cold start benchmarks

compilation service is enabled by default
```
./target/debug/benchmark_cold_start_large  spawned
```
can be disabled with flag
```
./target/debug/benchmark_cold_start_large spawned --component-compilation-disabled
```

Reply:
So we are back to having 2 more required